### PR TITLE
#426 Assertioncounter fix

### LINKED ISSUE #426: [Redesign] Total of issues is too large, because it counts all assertions
This is the case for two metrics: “total to evaluate” and “total evaluated”.

I see two ways to fix this: 

- filter the `$assertions` object to take out doubles in terms of their `num`, so that for each `num` there is only one assertion, then look at the length of the resulting object
- establish somewhere what the total to evaluate is for each possible conformance target (WCAG 2.0 A, WCAG 2.0 AA, WCAG 2.0 AAA, WCAG 2.1 A, etc…)

### PR DESCRIPTION
**Pull request by mhansma96**
Assertioncounter fix - the amount of open assertions now updates according to filters. Also the subcategories update with it

fixes #426

### CODE REVIEW

**Comment by hidde** [comment]
Tested this in the preview, and I see an issue: changing the conformance target in step 2 does not affect totals, changing filter in step 4 does. This should be the other way around: changing target in step should impact the totals, checking/unchecking in step 4 should not.

Would you be able to check this? Thanks!

**Comment by mhansma96** [comment]
I see, I will look into it, but maybe do need a little bit of clarification. 
Since there are 2 sets of filters in this situation: the ones on step 2 and the ones on step 4. If the ones on step 2 filter the actual amount of assertions, what do the filters in step 4 add? Because using them also changes the amount of actual assertions on the screen.

**Comment by hidde** [comment]
In step 2, the conformance target is the “goal“ of this report. It sets set a scope, so if the user says “I will report on WCAG 2.1 A + AA” in step 2, there are 50 SCs that fall under that, so their total will always be 50.

If they get to step 4, they can choose to look only at Level A or Level AA, but their goal is still A + AA (so the report will still show A + AA). 

Hope that helps.

File: src/components/ui/Auditor/Auditor.svelte [hunk 1]
```diff
@@ -56,16 +56,21 @@
...
+  if ($auditFilter['VERSION']) {
     $auditFilter['VERSION'] = [...$scopedWcagVersions].reverse().join();
     $auditFilter['LEVEL'] = CONFORMANCE_LEVELS.filter(
       (level) => $scopeStore['CONFORMANCE_TARGET'].indexOf(level) >= 0
     );
   }
 
-  $: criteriaFiltered = $wcagCriteria
+  $: if (filteredCriteria) {
+    console.log($scopeStore['CONFORMANCE_TARGET']);
```
Comment (hidde) [inline src/components/ui/Auditor/Auditor.svelte:69]: zelfde als hierboven, deze console logs mogen denk ik weg

File: src/components/ui/YourReport.svelte [hunk 2]
```diff
@@ -39,6 +39,14 @@
 
   import { wcag, scopedWcagVersions } from '@app/stores/wcagStore.js';
   import assertions from '@app/stores/earl/assertionStore/index.js';
+
+  import { CriteriaSelected } from '@app/stores/selectedCriteriaStore.js';
+  let criteriaCount = 0;
+  $: criteriaCount = $CriteriaSelected.length;
+
+  $: if (true) {
+    console.log(CriteriaSelected);
```
Comment (hidde) [inline src/components/ui/YourReport.svelte:48]: zelfde als boven

**Review by hidde — CHANGES_REQUESTED** [review]
Ziet er allemaal goed uit, dank! 

Zag nog een aantal console logs, die mogen er wmb uit.

**Review by hidde — APPROVED** [review]

**Pull request merged**